plugins/module_utils/role.py

```diff
@@ -31,20 +35,22 @@ def from_hypercore(cls, hypercore_data):
             name=hypercore_data["name"],
         )
 
-    def to_hypercore(self):
+    def to_hypercore(self) -> None:
         pass
 
-    def to_ansible(self):
+    def to_ansible(self) -> TypedDNSConfigToAnsible:
         return dict(
             uuid=self.uuid,
             name=self.name,
         )
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
         """
         One User is equal to another if it has ALL attributes exactly the same.
         This method is used only in tests.
         """
+        if not isinstance(other, Role):
```

Comment:
@domendobnikar why is this needed? Isn't enough just to put "other: Role". Anyway Role class is part of User module, where I need to add the type annotation, so I am also doing this in parallel with you, thus the comments...

Reply:
This is specifically for `__eq__` methods. Since `__eq__` is expecting an object of any type and by hinting only `Role` you are trying to override that supertype so MyPy won't let you override it. Right now this works in code because you know you are only checking Role vs Role, but MyPy doesn't know that hence the `NotImplemented` part.

Reply:
Aaa ok ok, thanks for the explanation!